sklearn/gaussian_process/kernels.py

```diff
@@ -0,0 +1,940 @@
+            contained subobjects that are estimators.
+
+        Returns
+        -------
+        params : mapping of string to any
+            Parameter names mapped to their values.
+        """
+        params = dict(kernel=self.kernel, exponent=self.exponent)
+        return params
+
+    @property
+    def theta(self):
+        return self.kernel.theta
+
+    @theta.setter
+    def theta(self, theta):
+        self.kernel.theta = theta
+
+    @property
+    def bounds(self):
+        return self.kernel.bounds
+
+    @bounds.setter
+    def bounds(self, bounds):
+        self.kernel.bounds = bounds
+
+    def __eq__(self, b):
+        return (self.kernel == b.kernel and self.exponent == b.exponent)
+
+    def __call__(self, X, Y=None, eval_gradient=False):
+        """ Return the kernel k(X, Y) and optionally its gradient.
+
+        Parameters
+        ----------
+        X : array, shape (n_samples_X, n_features)
+            Left argument of the returned kernel k(X, Y)
+
+        Y : array, shape (n_samples_Y, n_features), (optional, default=None)
+            Right argument of the returned kernel k(X, Y). If None, k(X, X)
+            if evaluated instead.
+
+        eval_gradient : bool (optional, default=False)
+            Determines whether the gradient with respect to the kernel
+            hyperparameter is determined.
+
+        Returns
+        -------
+        K : array, shape (n_samples_X, n_samples_Y)
+            Kernel k(X, Y)
+
+        K_gradient : array (opt.), shape (n_samples_X, n_samples_X, n_dims)
+            The gradient of the kernel k(X, X) with repect to the
+            hyperparameter of the kernel. Only returned when eval_gradient
+            is True.
+        """
+        if eval_gradient:
+            K, K_gradient = self.kernel(X, Y, eval_gradient=True)
+            K_gradient *= \
+                self.exponent * K[:, :, np.newaxis] ** (self.exponent - 1)
+            return K ** self.exponent, K_gradient
+        else:
+            K = self.kernel(X, Y, eval_gradient=False)
+            return K ** self.exponent
+
+    def __repr__(self):
+        return "{0} ** {1}".format(self.kernel, self.exponent)
+
+    def is_stationary(self):
+        """ Returns whether the kernel is stationary. """
+        return self.kernel.is_stationary()
+
+
+class ConstantKernel(Kernel):
+    """ Constant kernel.
+
+    Can be used as part of a product-kernel where it scales the magnitude of
+    the other factor (kernel) or as part of a sum-kernel, where it modifies
+    the mean of the Gaussian process.
+
+    Tunable kernel parameters
+    -------------------------
+    value : float
+        The constant value used for determining the magnitude (product-kernel)
+        or offset of mean (sum-kernel).
+    """
+
+    @classmethod
+    def from_literal(cls, literal):
+        if np.iterable(literal):
+            if len(literal) == 1:
+                return cls(literal[0])
+            elif len(literal) == 2:
```

Comment:
Wouldn't `from ... import ConstantKernel as C` suffice to make a shorthand for constant kernels? One would then have `C(.5, 0, 1) * other_kernel` instead of `(.5, 0, 1) * other_kernel`. I am still not convinced this feature is useful.